src/lib/replication/repl_dev/raft_state_machine.cpp

```diff
@@ -320,8 +328,16 @@ int RaftStateMachine::read_logical_snp_obj(nuraft::snapshot& s, void*& user_ctx,
 
 void RaftStateMachine::save_logical_snp_obj(nuraft::snapshot& s, ulong& obj_id, nuraft::buffer& data, bool is_first_obj,
                                             bool is_last_obj) {
+    if ((obj_id & snp_obj_id_type_mask) == 0) {
+        // Homestore preserved msg
+        if (m_rd.apply_snp_resync_data(data)) {
+            obj_id = snp_obj_id_type_mask;
```

Comment:
I am afraid I don't quite follow the logic here, what is the purpose of this line to set obj_id to this type_mask, by reading the log, it says it will be next obj id, but now sure how we achieved that by assigning to the type mask?

Reply:
okay, should we rename `snp_obj_id_type_mask` to `snp_obj_id_type_app`, which seems to be the case here right?

Reply:
Ok, `snp_obj_id_type_app` is clearer.